### ISSUE TITLE
"Ignoring sysex message found, not implemented: " with BlackMateria soundbank

### ISSUE THREAD

**Issue by p-phillips** [christofmuc/KnobKraft-orm#92]
Hi

I've built this under Ubuntu 20.04 and it connects to my prophet 12 correctly - I can download patches etc.

I have a set of Prophet 12 syx files from BlackMateria (https://soundcloud.com/black_materia/sets/patch-demos-p12). When I try to load the syx files into KnobKraft-orm, I get the following error message:

_Ignoring sysex message found, not implemented:_

and the soundbank doesn't load up.

Any idea what might be the cause of this ?  

Pete

**Comment by p-phillips** [comment]
Hi.

Here you go.  first 20 bytes for the first 10 lines.

Fingers crossed this helps.
Pete

```
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 00 00 24 24 24 24 32 41 32 00 41 48 00 00 00
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 01 00 24 24 24 24 2d 3a 28 00 42 53 4b 53 53
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 02 00 18 18 18 18 34 3c 2f 00 2e 53 35 53 53
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 03 00 18 18 18 18 37 37 32 00 2d 61 62 00 62
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 04 00 18 18 18 18 37 2e 32 00 32 64 64 00 00
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 05 00 18 18 24 24 39 2d 3b 00 29 64 64 64 64
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 06 00 24 24 18 18 2b 39 2f 00 35 57 57 57 57
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 07 00 24 24 18 18 2b 39 2f 00 35 57 57 57 57
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 08 00 18 18 18 18 32 37 2d 00 32 00 62 62 00
20:07:39: Ignoring sysex message found, not implemented: f0 01 2b 02 00 09 00 24 30 30 24 32 2b 39 00 32 60 16 16 00

```

**Comment by christofmuc** [comment]
Ok, I can confirm that this is related to the Prophet 12 Desktop module using 2b as an identifier and not 2a. So far we had seen the 2b only in device detect response, but it seems it also can produce it on download of patches.

I attach a new version of the Prophet 12 adapation here - you can download the Python file, unzip it and drop it in the custom adaptations folder (directory can be set with the button in the setup tab). When you then restart the Orm, it will use this new version (there will be a log message in the log view stating that it is indeed overriding the built-in adaptation, something like " Overriding built-in adaptation DSI Prophet 12 (found in user directory D:\Development\github\KnobKraft-Orm\adaptions)")

If you confirm this works, I'll make the new version part of the future new releases, you can then delete this override version when you get a new release with the fix built in. 
[DSI Prophet 12.zip](https://github.com/christofmuc/KnobKraft-orm/files/6472345/DSI.Prophet.12.zip)

(Note to myself: This already came up in #13 but seemingly wasn't fixed properly)

**Comment by p-phillips** [comment]
So.....

This now can load the patches from the sysex file, and it can read the patches in the prophet bank.

```
19:39:49: Adaptation: Warning: file name DSI Prophet 12 is not a valid module identifier in Python, please use only lower case letters and numbers
19:39:50: Overriding built-in adaptation DSI Prophet 12 (found in user directory /home/pete/src/KnobKraft-orm/KnobKraft-orm/custom-adaption)
19:39:50: Lost communication with DSI Prophet 12 on channel 2 of device  - please rerun auto-detect synths!
19:39:50: Turning off USB input Electra Controller
19:40:06: Skipping patch Time 2 Develop because it is a duplicate of Time 2 Develop
19:40:06: Skipping patch Gentle Hybrid because it is a duplicate of Gentle Hybrid
19:40:06: Skipping patch Analogical String because it is a duplicate of Analogical String
19:40:06: Retrieved 96 new or changed patches from the synth, uploaded to database
19:40:11: Sending patch Lush Prophet XII
19:40:45: Sending patch Jupiter Lush

```
However, when I click on a patch in the interface (Lush Prophet XII for example), it does not transmit to the synth (previously, it would read the banks from the synth and clicking on any patch made that the live patch on the prophet).

Pete

**Comment by christofmuc** [comment]
@p-phillips Hi Pete, somehow it did not detect your Prophet 12, which is why it would not send (as it doesn't know where to). Can you run the autodetection again (MIDI menu)? It should detect the Prophet 12 and show it with a green bar under the Prophet 12 button at the top.

**Comment by p-phillips** [comment]
Ah. That works perfectly.  Brilliant stuff.  Thanks for that. 

Pete

**Comment by christofmuc** [comment]
New version of adaptation was released with KnobKraft 1.12.3
